apps/omg_utils/lib/omg_utils/metrics.ex

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Utils.Metrics do
+  @moduledoc """
+  In option can discard some metrics trigger.
+  ## Example
+
+      config :metrics,
+        discard: [:State]
+  """
+
+  use Appsignal.Instrumentation.Decorators
+
+  use Decorator.Define,
+    measure_start: 0,
+    measure_event: 0
+
+  @discard_namespace_metrics Application.get_env(:metrics, :discard, [])
+
+  def measure_start(body, context) do
+    namespace = context.module |> Module.split() |> List.last() |> String.to_existing_atom()
+
+    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace)),
+      do: body,
+      else: Appsignal.Instrumentation.Decorators.transaction(namespace, body, context)
```

Comment:
what about my comment on the namespaces/event groups? 

> I'm still thinking about the naming convention for the event groups and namespaces. I think there's a problem with camel-case entries like LevelDB as they suggest a module, while there is no such module strictly speaking.
> 
> Perhaps we should consider using:
> 1/ __MODULE__ for event groups always, consistently
> 2/ something more top-level for transactions' namespaces? web/background is the appsignals default, which I agree is pretty coarse, but maybe it will we what is most useful actually thinking ?
> 
> We also need to take AppSignal's UX into account - switching namespaces filters is very clunky, so the less the better IMHO

are we dead set to have such wealth of namespaces? what are the benefits

Reply:
We only use the last parts from the module name. (appsignal does not let you use a dot)
appsignal default namespace for transaction is http_request.

We are limited to names that can be found in atoms, 
without string pattern (:"any name" ps. module name is :"Elixir.module_name")

We can still find replacement signs for dots, e.g. ```@```

Reply:
again: why not use just `background` vs `http_request`, the two default namespaces AppSignal provides? The question was "what are the benefit of using your multiple custom namespaces?", nothing about dots.

and where is the example result to look at?

Reply:
![image](https://user-images.githubusercontent.com/429892/57211291-00e6e480-6fe0-11e9-83e8-1932a5ad98aa.png)
when we use custom names, 
we have more information (lock at ExitProcessor, from BlockGetter we use 39.69% and from EthereumEventListener only 1.09%). 
more precise cause of usage

Reply:
OK, I'm not sure which is better, let's just proceed like this and try in practice, unless anyone has further objections.

I suggest to leave a comment inside the moduledoc:

```
NOTE: the namespace and event group naming convention here is tentative.
It is possible we'll revert to standard coarser division into `web` and `background` namespaces Appsignal suggests
```